source/extensions/tracers/xray/util.cc

```diff
@@ -0,0 +1,166 @@
+#include "extensions/tracers/xray/util.h"
+
+#include <algorithm>
+#include <chrono>
+#include <random>
+#include <regex>
+#include <vector>
+
+#include "common/common/hex.h"
+#include "common/common/utility.h"
+
+#include "rapidjson/document.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+void Util::mergeJsons(std::string& target, const std::string& source,
```

Comment:
Seems like this one is similar to zipkin's? https://github.com/envoyproxy/envoy/blob/7ddd7657a11bc58c814c086311d655df2a9824e1/source/extensions/tracers/zipkin/util.cc#L19 and also the `addArrayToJson` one. Can we have a common code to include between tracers here?

Reply:
Yes, there are similar. However, I didn't implemented Zipkin tracer. I might not have enough time to also test whether it breaks zipkin or not after the common code implementation is done.